cmd/geth/chaincmd.go

```diff
@@ -655,7 +655,7 @@ func dump(ctx *cli.Context) error {
 	if err != nil {
 		return err
 	}
-	triedb := utils.MakeTrieDatabase(ctx, db, true, false) // always enable preimage lookup
```

Comment:
Why not readonly mode? It's dangerous.

Reply:
Original code is not read-only mode. I change it into read-only mode.